### PR TITLE
audit: Port audit_conflicts method to rubocop and add tests

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----
#569 
This PR partially ports `audit_conflicts` method to a custom cop. Remaining code in `audit_conflicts` uses Homebrew internals methods and I could not figure out their exact functionality to port them. Please let me know if this approach is not good.

### CODE REVIEW

File: Library/Homebrew/rubocops/conflicts_cop.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require_relative "./extend/formula_cop"
+require_relative "../extend/string"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits versioned Formulae for `conflicts_with`
+      class Conflicts < FormulaCop
+        MSG = <<-EOS.undent
+            Versioned formulae should not use `conflicts_with`.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/conflicts_cop.rb]: Only indent by two characters here.

File: Library/Homebrew/rubocops/conflicts_cop.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            Use `keg_only :versioned_formula` instead.
+        EOS
+
+        def audit_formula(_node, _class_node, _parent_class_node, body)
+          return unless versioned_formula?
+          whitelist = %w[node@ bash-completion@].freeze
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/conflicts_cop.rb]: Let's make this a constant earlier in the class and have one formula per line so they stand out a bit more

File: Library/Homebrew/rubocops/conflicts_cop.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    module FormulaAudit
+      # This cop audits versioned Formulae for `conflicts_with`
+      class Conflicts < FormulaCop
+        MSG = <<-EOS.undent
+          Versioned formulae should not use `conflicts_with`.
+          Use `keg_only :versioned_formula` instead.
+        EOS
+
+        WHITELIST = %w[node@
+                       bash-completion@].freeze
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/conflicts_cop.rb]: Let's go for:

```ruby
WHITELIST = %w[
   node@
   bash-completion@
].freeze
```

Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/conflicts_cop.rb]: that will reduce the diff on future additions here.

Comment (GauthamGoli) [inline Library/Homebrew/rubocops/conflicts_cop.rb]: Waoh. Never thought of that!

**Comment by GauthamGoli** [comment]
@MikeMcQuaid Moving the two private methods defined in `conflicts_cop.rb` to `formula_cop.rb` as they can be reused in other cops. ( `conflicts_cop.rb`)

**Pull request merged**